modules/raft/src/main/java/org/apache/ignite/internal/raft/server/RaftGroupOptions.java

```diff
@@ -42,6 +43,9 @@ public class RaftGroupOptions {
     /** Configuration of own striped disruptor for FSMCaller service of raft node, {@code null} means use shared disruptor. */
     private @Nullable RaftNodeDisruptorConfiguration ownFsmCallerExecutorDisruptorConfig;
 
+    /** Marshaller to marshall/unmarshall commands. */
+    private @Nullable Marshaller commandsMarshaller;
```

Comment:
Why do we need to override marshaller ? For test purposes ?

Reply:
For partitions, we need to quickly extract requiredCatalogVersion from command's serialized representation without parsing the whole command. To do so, I tweaked the marshaller used for partitions, but it adds one byte to each command that does not have that requiredCatalogVersion. If we use same marshaller for other (non-partition) groups, we unnecessarily increase their (serialized) sizes as well. Hence the necessity to have ability to supply a different marshaller.